odoo/odoo
# [FIX] html_editor, mail: adds a way to mention (@, #)

The purpose of this commit is to reintroduce a way to mention a "partner" or a "channel" in the HtmlComposerMessageField.
So you will be able to mention a partner by typing "@" and a channel by typing "#"".

This commit will also fix an autofocus problem in the editor overlay when you're in a dialog.

Problem:
    Autofocus in an overlay is not applied when in a dialog.

Reason:
    Autofocus is only applied if the component is in the activeElement
    of the ui. When we are in a dialog, it is the activeElement but the
    overlays are not in it.

Solution:
    Offer an api point allowing an overlay to become the activeElement.

How to reproduce:
- Go to a form view in a dialog with an HtmlComposerMessageField
- type "@"" in the html field

Before this commit:
    The focus remains in the editor

After this commit:
    The focus is moved to the MentionList input.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:




---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr
